pkg/network/ebpf/c/protocols/events-types.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __USM_EVENTS_TYPES_H
+#define __USM_EVENTS_TYPES_H
+
+#include <linux/types.h>
+
+#define BATCH_BUFFER_SIZE (4*1024)
```

Comment:
the size of this doesn't matter that much for system-probe memory usage because you should have only a few "live" instances of this in memory at any given point in time because the memory backing up these types is pooled
https://github.com/DataDog/datadog-agent/blob/0ae31014e9e8317ed11b73c5b98acc088b903e69/pkg/ebpf/perf.go#L51-L53

to your second point, I think that's a totally valid concern, but I think I'd prefer to leave this to a separate PR if this becomes a real issue. Here I'm assuming that 4KB should be a "good enough" size to batch a decent amount of events for most use-cases we'll have. alternatively we could also have a batch type for each USM integration instead of a shared one, but I'm afraid this could make the userspace code even more trickier which is why I opted for a one-size fits all solution for the moment.